Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1552,7 +1553,19 @@ void FrameLoader::loadURL(FrameLoadRequest&& frameLoadRequest, const String& ref
     action.setShouldReplaceDocumentIfJavaScriptURL(frameLoadRequest.shouldReplaceDocumentIfJavaScriptURL());
     action.setIsInitialFrameSrcLoad(frameLoadRequest.isInitialFrameSrcLoad());
     action.setNewFrameOpenerPolicy(frameLoadRequest.newFrameOpenerPolicy());
-    action.setNavigationAPIType(determineNavigationType(newLoadType, frameLoadRequest.navigationHistoryBehavior()));
+    auto historyHandling = frameLoadRequest.navigationHistoryBehavior();
+    bool isSameOrigin = frameLoadRequest.requesterSecurityOrigin().isSameOriginDomain(frame->document()->securityOrigin());
+    if (historyHandling == NavigationHistoryBehavior::Auto && !isReload(newLoadType)) {
+        RefPtr document = m_frame->document();
```

Comment:
Not sure why we are doing this here and not a couple lines earlier.

Reply:
To be clear, that’s because `frame->document()` is used in `isSameOrigin`, so why wouldn’t we use the local variable for that too.

Reply:
I see what you mean, will try to make a clean up PR at some point.